lib/matplotlib/tight_layout.py

```diff
@@ -169,19 +169,34 @@ def auto_adjust_subplotpars(fig, renderer,
     if not margin_left:
         margin_left = max([sum(s) for s in hspaces[::cols + 1]] + [0])
         margin_left += pad_inches / fig_width_inch
+        # bugfix for bug-#5456
+        # simple check for the error
+        if margin_left > 0.5:
+            margin_left = 0.05
```

Comment:
Why these magic numbers?  It might be more reliable to do a check for the actual fail condition (`margin_left > 1 - margin_right`) at the end and fix it there once you know all of the margins.